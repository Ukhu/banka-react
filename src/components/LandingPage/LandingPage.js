import React from 'react';

const LandingPage = () => (
  <div>
    <h1>Hello</h1>
  </div>
);

export default LandingPage;
