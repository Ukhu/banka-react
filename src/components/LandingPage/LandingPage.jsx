import React from 'react';
import { Link } from 'react-router-dom';
import UnauthenticatedHeader from '../common/Header/UnauthenticatedHeader';
import Footer from '../common/Footer/Footer';

const LandingPage = () => {
  const user = localStorage.getItem('user');

  return (
    user ? window.history.back()
      : (
        <>
          <UnauthenticatedHeader />
          <main>
            <section id="landing-page">
              <div id="overlay">
                <div className="container">
                  <div className="overlay-content">
                    <h1>Welcome To Banka</h1>
                    <p>
                  Your number one banking services app. Sign Up with us today to enjoy
                  the numerous benefits we offer
                    </p>
                    <Link to="/signup"><button type="button" className="signup-btn">Sign Up</button></Link>
                    <button type="button" className="login-btn">Login</button>
                  </div>
                </div>
              </div>
            </section>
          </main>
          <Footer />
        </>
      )
  );
};

export default LandingPage;
