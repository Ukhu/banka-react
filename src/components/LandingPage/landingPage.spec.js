import React from 'react';
import { configure, shallow } from 'enzyme';

import Adapter from 'enzyme-adapter-react-16';
import LandingPage from './LandingPage';

configure({ adapter: new Adapter() });

describe('LandingPage component <LandingPage />', () => {
  it('renders correctly', () => {
    const wrapper = shallow(
      <LandingPage />,
    );

    expect(wrapper).toMatchSnapshot();
  });

  it('has the header and footer components', () => {
    const wrapper = shallow(
      <LandingPage />,
    );

    expect(wrapper.find('UnauthenticatedHeader')).toHaveLength(1);
    expect(wrapper.find('Footer')).toHaveLength(1);
  });

  it('has two buttons', () => {
    const wrapper = shallow(
      <LandingPage />,
    );

    expect(wrapper.find('button')).toHaveLength(2);
  });
});
