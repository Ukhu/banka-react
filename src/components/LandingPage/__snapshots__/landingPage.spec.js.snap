// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LandingPage component <LandingPage /> renders correctly 1`] = `ReactWrapper {}`;
